src/plugins/search/search.js

```diff
@@ -165,13 +165,18 @@ export function search(query) {
             end = postContent.length;
           }
 
-          const matchContent =
-            '...' +
-            escapeHtml(postContent)
-              .substring(start, end)
-              .replace(regEx, `<em class="search-keyword">${keyword}</em>`) +
-            '...';
-
+          let content = escapeHtml(postContent).substring(start, end);
+          let matchKeywords = content.match(regEx);
+
+          matchKeywords = new Set(matchKeywords);
+          matchKeywords.forEach(keyword => {
+            content = content.replace(
+              new RegExp(keyword, 'g'),
+              `<em class="search-keyword">${keyword}</em>`
+            );
+          });
+
+          const matchContent = '...' + content + '...';
```

Comment:
when the title matched keyword, `new Set(matchKeywords)` will filte the `matchKeywords  == null`.

when the `postContent` has more than one matched and the keyword has the same case, like testCase2.
```md
search a keyword testDemo, and testDemo have more than one in `postContent`. 
```
Using `new Set(matchKeywords)` to remove the duplicated replaces from `testDemo` to ``<em class="search-keyword">testDemo</em>``.